youtube_dl/extractor/npo.py

```diff
@@ -478,6 +478,80 @@ class HetKlokhuisIE(NPODataMidEmbedIE):
     }
 
 
+class NPORecentsIE(NPOIE):
+    IE_NAME = 'npo.nl:recents'
+    npo12_regex = r"""<div class='span4'>\s*<div class='image-container'>\s*<a href="(.*?)">\s*(<div class="program-not-available">)?"""
+    npo3_regex = r"""<div class='span4 image'>\s*<a href="(.*?)">\s*<div class="meta-container">\s*<div class="meta first">\s*<div class="md-label"><span class="npo-glyph triangle-right"></span></div>\s*<div class="md-value">.*?</div>\s*</div>\s*</div>\s*(<div class="program-not-available">)?"""
+    _VALID_URL = r'(?:https?://)?(?:www\.)?npo\.nl/(?P<alt_id>[^/]+)/(?P<program_id>\w+_\d+)'
+    _TESTS = [{
+        # Example of an npo3 program
+        'url': 'https://www.npo.nl/keuringsdienst-van-waarde/KN_1678993',
+        'info_dict': {
+            'title': 'Keuringsdienst van Waarde',
+            'id': 'KN_1678993',
+            'description': 'md5:5ffaf131f175d8a771e7a7884833dad2'
+        },
+        'playlist_mincount': 8
+    }, {
+        # Example of an npo1/npo2 program
+        'url': 'https://www.npo.nl/jinek/KN_1676589',
+        'info_dict': {
+            'title': 'Jinek',
+            'id': 'KN_1676589',
+            'description': 'md5:6998986899b4903395f0cdd0670cedaf'
+        },
+        'playlist_mincount': 8
+    }, {
+        # Example of a program for which there will be only one available episode (if any)
+        'url': 'https://www.npo.nl/midsomer-murders/POW_00828660',
+        'info_dict': {
+            'title': 'Midsomer murders',
+            'id': 'POW_00828660',
+            'description': 'md5:a8b6e9d3e3bd367be88766e3ce8e8362'
+        },
+        'playlist_maxcount': 1
+    }]
+
+    def _extract_entries(self, webpage, program_id, program_url):
+        is_npo3 = 'www-assets.npo.nl/uploads/tv_channel/265/logo/smaller_npo3-logo.png' in webpage
+
+        if is_npo3:
+            episodes_url = '%s//search?category=broadcasts&page=1' % program_url
```

Comment:
There are more pages. All should be extracted and extractor should be renamed to `NPOProgramIE`.